backend/chainlit/auth/cookie.py

```diff
@@ -76,26 +76,60 @@ async def __call__(self, request: Request) -> Optional[str]:
         return token
 
 
+def reconstruct_token_from_cookies(request_cookies: dict) -> Optional[str]:
+    """
+    Read all chunk cookies and reconstruct the token
+    """
+
+    # Gather all auth_chunk_i cookies, sorted by their index
+    chunk_parts = []
+    i = 0
+    while True:
+        cookie_key = f"{_auth_cookie_name}_{i}"
+        if cookie_key not in request_cookies:
+            break
+        chunk_parts.append(request_cookies[cookie_key])
+        i += 1
+
+    joined = "".join(chunk_parts)
+
+    return joined if joined != "" else None
+
+
 def set_auth_cookie(response: Response, token: str):
```

Comment:
We should be careful here to ensure the case where we're overwriting cookies and not deleting old ones.

Imagine:
1. `set_auth_cookie(r, 'incredibly_<...>_long_token')` sets `access_token_0`, `access_token_1`.
2. `set_auth_cookie(r, 'more_reasonable_token')` sets only `access_token_0`.
3. `get_token_from_cookies(🍪)` returns the concatenation of the new `access_token_0` and the _old_ `access_token_1`.